src/test/kotlin/ecommerce/controller/admin/AdminProductControllerTest.kt

```diff
@@ -0,0 +1,170 @@
+package ecommerce.controller.admin
+
+import ecommerce.dto.auth.LoginRequest
+import ecommerce.dto.products.ProductDTO
+import ecommerce.dto.products.ProductPatchDTO
+import ecommerce.mapper.UserRowMapper
+import ecommerce.repository.ProductRepository
+import ecommerce.service.AdminAuthService
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.jdbc.core.JdbcTemplate
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class AdminProductControllerTest {
+    private lateinit var token: String
+
+    @Autowired
+    private lateinit var productRepository: ProductRepository
+
+    @Autowired
+    private lateinit var jdbcTemplate: JdbcTemplate
+
+    @Autowired
+    private lateinit var userRowMapper: UserRowMapper
+
+    @Autowired
+    private lateinit var adminAuthService: AdminAuthService
+
+    @BeforeEach
+    fun init() {
+        val sql = "select * from users where role = 'ADMIN'"
+        val result = jdbcTemplate.query(sql, userRowMapper).first()
+        token =
+            adminAuthService.signIn(
+                LoginRequest(
+                    result.email, result.password,
+                ),
+            )
```

Comment:
I think it should be fetched by id; otherwise, the result could contain multiple items 🙂